lib/content-repo.js

```diff
@@ -13,14 +13,23 @@ module.exports = {
      * @param {Number} height The desired height of the thumbnail.
      * @param {string} resizeMethod The thumbnail resize method to use, either
      * "crop" or "scale".
+     * @param {Boolean} allowDirectLinks If true, return any non-mxc URLs
+     * directly. Fetching such URLs will leak information about the user to
+     * anyone they share a room with. If false, will return the emptry string
+     * for such URLs.
      * @return {string} The complete URL to the content.
      */
-    getHttpUriForMxc: function(baseUrl, mxc, width, height, resizeMethod) {
+    getHttpUriForMxc: function(baseUrl, mxc, width, height,
+                               resizeMethod, allowDirectLinks) {
```

Comment:
Can we invert the flag here so we don't make a breaking API change?

Existing clients will have `allowDirectLinks == undefined == false` which means it will return an empty string rather than what they put in on lines `:27-32`. Inverting the flag to be `disallowDirectLinks` will keep existing behaviour, and still allow you to block any non-MXC URL.